topics/src/POO/Ex2/PostgreSQLConnectionFactory.java

```diff
@@ -0,0 +1,8 @@
+package POO.Ex2;
+
+public class PostgreSQLConnectionFactory implements ConnectionFactory {
```

Comment:
As Factory Pattern definition, you should be able to create an object instance of a family using the Factory, for example

```suggestion
public final class ConnectionFactory {
    private ConnectionFactory() { }
    public static Connection getConnection(String type) {
        switch(type) {
            "java": return Java.getInstance();
            "mysql": return Mysql.getInstance();
            ...
            default: return NoOp.getInstance();
        }
    }
}
```

Reply:
Could we having a misunderstanding? this is the second point of the practice :
"Create an abstract factory example for the different type of SQL connections." 
this code that you showed me, isn't a Factory pattern? not AbstractFactory or am i wrong?

Reply:
this are the links from where i reading
https://www.oodesign.com/factory-method-pattern.html
https://www.oodesign.com/abstract-factory-pattern.html

Reply:
wee you could also refer to https://www.oodesign.com/factory-pattern.html and there is a producId, but that product id could be anything that make sense in your context